lib/providers/phabricator.js

```diff
@@ -0,0 +1,45 @@
+// Load modules
+var Joi = require('joi');
+var Hoek = require('hoek');
+
+// Declare internals
+
+var internals = {};
+
+internals.schema = Joi.object({
+    uri: Joi.string().uri().required()
+}).required();
+
+exports = module.exports = function (options) {
+
+    var results = Joi.validate(options, internals.schema);
+
+    Hoek.assert(!results.error, results.error);
+
+    var settings = results.value;
+
+    return {
+        protocol: 'oauth2',
+        auth: settings.uri + '/oauthserver/auth/',
+        token: settings.uri + '/oauthserver/token/',
+        scope: ['whoami'],
+        scopeSeparator: ',',
+        profile: function (credentials, params, get, callback) {
+            var query = {
```

Comment:
we add a newline after function declarations.  please update the PR with the newlines. Here is more detail on the styleguide: hapijs.com/styleguide